src/app/pet-profile/pet-form/pet-form.component.ts

```diff
@@ -0,0 +1,215 @@
+import { ChangeDetectionStrategy, Component, signal, OnInit, Input, DestroyRef, inject } from '@angular/core';
+import { CardComponent } from "../../ui/card/card.component";
+import { FormArray, FormControl, FormGroup, FormsModule, ReactiveFormsModule, Validators } from '@angular/forms';
+import { MatButtonToggleModule } from '@angular/material/button-toggle';
+import { Pet_Form_DATA, Pet_Profile } from '../models/pet-profile.model';
+import { MatIconModule } from '@angular/material/icon';
+import { MatFormFieldModule } from '@angular/material/form-field';
+import { MatInputModule } from '@angular/material/input';
+import { MatDatepickerModule } from '@angular/material/datepicker';
+import { MatNativeDateModule } from '@angular/material/core';
+import { medicationValidator } from './pet-form-validators';
+import { CustomInputComponent } from "../../ui/custom-input/custom-input.component";
+import { CustomSelectionComponent, SELECTION } from '../../ui/custom-selection/custom-selection.component';
+
+@Component({
+  selector: 'app-pet-form',
+  imports: [
+    CardComponent,
+    FormsModule,
+    ReactiveFormsModule,
+    MatButtonToggleModule,
+    MatIconModule,
+    MatFormFieldModule,
+    MatInputModule,
+    MatDatepickerModule,
+    MatNativeDateModule,
+    CustomInputComponent,
+    CustomSelectionComponent
+],
+  templateUrl: './pet-form.component.html',
+  styleUrl: './pet-form.component.scss',
+  changeDetection: ChangeDetectionStrategy.OnPush,
+})
+export class PetFormComponent implements OnInit {
+  private destroyRef = inject(DestroyRef);
+
+  title = signal('Pet Profile');
+  @Input() pet!: Pet_Profile;
+  petProfileForm!: FormGroup;
+
+  goalSelection : SELECTION[] = [
+    { value: 'Maintain' },
+    { value: 'Lose' },
+    { value: 'Gain' },
+  ]
+ 
+
+  ngOnInit(): void {
+    // Setup initial form values
+    const initialPetProfile: Pet_Form_DATA = {
+      id: this.pet?.id ?? null,
```

Comment:
Can you just ensure that pet and all of these properties exist before instantiation of initialPetProfile? If not, can you use a partial?

e.g.
```ts
partial<Pet_Form_DATA>
```